src/makeInfinityAdder.js

```diff
@@ -1,12 +1,24 @@
 'use strict';
 
-/***
- *
- * @return {function}
- */
-
+/**
+ * @return {function} */
 function makeInfinityAdder() {
-  // write code here
+  const augend = [0, 0];
+
+  const summer = (addend) => {
+    if (addend !== undefined) {
+      augend[0] += addend;
+
+      return summer;
+    }
+
+    augend[1] = augend[0];
```

Comment:
we can just create some var instead of this

Reply:
creation extra variables increase stress to system. more efficiently make buffer.

Reply:
I think creating two variables ahead of time will be much better and clearer than one array

Reply:
where is there operation of array's clear? only assignment. buffer will be create one time and used every time for calculation, but now shell needs to create 'tempSum' every time even it won't be used.

Reply:
agree with @etojeDenys. Creating an array is unnecessary here